### ISSUE TITLE
Explanation of tf.keras.layers.CategoryEncoding output_mode='multi_hot' behavior

### ISSUE THREAD

**Issue by oonisim** [tensorflow/tensorflow#52892]
## URL(s) with the issue:
https://www.tensorflow.org/api_docs/python/tf/keras/layers/CategoryEncoding

## Description of issue (what needs changing):

Need clarification on the statement below about multi hot encoding.

> "multi_hot": Encodes each sample in the input into a single array of num_tokens size, containing a 1 for each vocabulary term present in the sample. Treats the last dimension as the sample dimension, if input shape is (..., sample_length), output shape will be (..., num_tokens).

## Clarification

Please help understand the definition of **multi hot encoding** of tf.keras.layers.CategoryEncoding and the behavior of ```output_mode='multi_hot'```.

### Background 
According to [What exactly is multi-hot encoding and how is it different from one-hot?](https://stats.stackexchange.com/a/467672):

> If you would use multi-hot-encoding you would first label-encode your classes, thus having only a single number which represents the presence of a class (e.g. 1 for 'dog') and then convert the numerical labels to binary vectors of size log2(5)=3.  
> Examples:
> ```
> 'cat'  = [0,0,0]  
> 'dog'  = [0,0,1]  
> 'fish' = [0,1,0]  
> 'bird' = [0,1,1]  
> 'ant'  = [1,0,0]   
> ```

### Behaviour of [tf.keras.layers.CategoryEncoding][2]

The document says ```num_tokens``` is the total number of tokens the layer should support. 

> ### args
> #### num_tokens
> The total number of tokens the layer should support. All inputs to the layer must integers in the range 0 <= value < num_tokens, or an error will be thrown.
> #### output_mode
> * "one_hot": Encodes each individual element in the input into an array of num_tokens size, containing a 1 at the element index. If the last dimension is size 1, will encode on that dimension. If the last dimension is not size 1, will append a new dimension for the encoded output.
> * "multi_hot": Encodes each sample in the input into **a single array of num_tokens size, containing a 1 for each vocabulary term present in the sample**. Treats the last dimension as the sample dimension, if input shape is (..., sample_length), output shape will be (..., num_tokens).

According to the definitions of multi hot encoding above, I expected ```tf.keras.layers.CategoryEncoding(num_tokens=5, output_mode="multi_hot")``` encodes 5 tokens into an array of size 3. 

However, the document says "multi_hot" encodes each sample into **a single array of num_tokens size**, containing a 1 for each vocabulary term present in the sample, and behaves as such.

```
dataset = tf.data.Dataset.from_tensor_slices(tf.constant(['cat', 'dog', 'fish', 'bird']))

lookup = tf.keras.layers.StringLookup(max_tokens=5, oov_token='[UNK]')
lookup.adapt(dataset)
lookup.get_vocabulary()
---
['[UNK]', 'fish', 'dog', 'cat', 'bird']

mhe = tf.keras.layers.CategoryEncoding(num_tokens=lookup.vocabulary_size(), output_mode="multi_hot")
print(f"cat: {mhe(lookup(tf.constant('cat'))).numpy()}")
print(f"dog: {mhe(lookup(tf.constant('dog'))).numpy()}")
---
cat: [0. 0. 0. 1. 0.]
dog: [0. 0. 1. 0. 0.]
```

Which has no difference from One Hot En coding.

```
ohe = tf.keras.layers.CategoryEncoding(num_tokens=lookup.vocabulary_size(), output_mode="one_hot")
print(f"cat: {ohe(lookup(tf.constant('cat'))).numpy()}")
print(f"dog: {ohe(lookup(tf.constant('dog'))).numpy()}")
---
cat: [0. 0. 0. 1. 0.]
dog: [0. 0. 1. 0. 0.]
```

For multi value inputs, multi_hot only handles the 1st value.
```
print(ohe(lookup(tf.constant(['cat', 'dog']))).numpy())
---
[[0. 0. 0. 1. 0.]
 [0. 0. 1. 0. 0.]]

print(mhe(lookup(tf.constant(['cat', 'dog']))).numpy())
---
[0. 0. 1. 1. 0.]
```

To handle multiple inputs, need to be 2D array.
```
print(mhe(lookup(tf.constant([['cat'], ['dog']]))).numpy())
---
[[0. 0. 0. 1. 0.]
 [0. 0. 1. 0. 0.]]
```

Apparently the definition of **mluti hot encoding** of ```tf.keras.layers.CategoryEncoding``` is not the same with the one in [What exactly is multi-hot encoding and how is it different from one-hot?](https://stats.stackexchange.com/a/467672).

  [1]: https://www.tensorflow.org/tutorials/structured_data/preprocessing_layers
  [2]: https://www.tensorflow.org/api_docs/python/tf/keras/layers/CategoryEncoding

**Comment by mohantym** [comment]
Hi @Saduf2019 ! Could you please look into this issue ?

**Comment by Saduf2019** [comment]
@oonisim 
The Document clearly says: Encodes each sample in the input into a single array of num_tokens size, containing a 1 for each vocabulary term present in the sample. Treats the last dimension as the sample dimension, if input shape is (..., sample_length), output shape will be (..., num_tokens)
